spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/pulsar/PulsarProperties.java

```diff
@@ -535,7 +535,7 @@ public static class Consumer {
 		 * Dead letter policy to use.
 		 */
 		@NestedConfigurationProperty
-		private final DeadLetterPolicy deadLetterPolicy = new DeadLetterPolicy();
+		private DeadLetterPolicy deadLetterPolicy;
```

Comment:
The `new DeadLetterPolicy()` has a zero for max retry count and the builder fails w/ that non-positive value. I have added a consumer path on the Integration test to make sure we pass our defeault props through an actual consumer builder as the mock does not do this validation.

Reply:
I've added an extra assert in our own mapper code as well to catch problems early.